packages/tldraw/src/lib/shapes/shared/useAsset.ts

```diff
@@ -17,22 +18,68 @@ export function useAsset(shapeId: TLShapeId, assetId: TLAssetId | null, width: n
 		shapeScale,
 	])
 
+	const didAlreadyResolve = useRef(false)
+
+	useEffect(() => {
+		if (url) didAlreadyResolve.current = true
+	}, [url])
+
 	useEffect(() => {
 		if (isCulled) return
 
+		if (assetId && !asset?.props.src) {
+			const preview = editor.getTemporaryAssetPreview(assetId)
+			if (preview) {
+				setUrl(preview.retain())
```

Comment:
a comment here would be good to describe the lifecycle. i think you and i know what's going on but reference counting is tricky and i think we should make this a little clearer to future devs on what is being mitigated here.

to be specific:
- that the `retain` here is to stop it being released from the uploader side of things (which initially kicked off the process). 
- that sometimes it's setting the RC and sometimes it's setting the string and how it eventually "settles"